drill-admin/src/main/kotlin/com/epam/drill/endpoints/plugin/DrillPluginWs.kt

```
@@ -82,16 +84,11 @@ class DrillPluginWs(override val kodein: Kodein) : KodeinAware, WsService {
                                 MessageType.SUBSCRIBE -> {
                                     val subscribeInfo = SubscribeInfo.serializer() parse event.message
                                     saveSession(event)
-                                    val buildVersion = subscribeInfo.buildVersion
 
-
-                                    val message =
-                                        eventStorage[
-                                                subscribeInfo.agentId + ":" +
-                                                        event.destination + ":" +
-                                                        if (buildVersion.isNullOrEmpty()) subscribeInfo.agentId
-                                                        else buildVersion
-                                        ]
+                                    val buildVersion = subscribeInfo.buildVersion ?:
+                                        agentManager[subscribeInfo.agentId]?.buildVersion ?: ""
```

I think in last chain u should throw an exception and close socket